cmd/fluent-watcher/fluentd/Dockerfile.amd64

```diff
@@ -8,54 +10,25 @@ RUN echo $(ls -al /code)
 RUN CGO_ENABLED=0 go build -ldflags '-w -s' -o /fluentd/fluentd-watcher /code/cmd/fluent-watcher/fluentd/main.go
 
 # Fluentd main image
-FROM alpine:3.20
-LABEL Description="Fluentd docker image" Vendor="Fluent Organization" Version="1.17.0"
+FROM ${BASE_IMAGE}:${BASE_IMAGE_TAG}
+LABEL Description="Fluentd docker image" Vendor="Fluent Organization" Version=${BASE_IMAGE_TAG}
+ENV TINI_VERSION=0.18.0
 
 # Do not split this into multiple RUN!
 # Docker creates a layer for every RUN-Statement
-# therefore an 'apk delete' has no effect
-RUN apk update \
- && apk add --no-cache \
-        ca-certificates \
-        ruby ruby-irb ruby-etc ruby-webrick \
-        tini \
- && apk add --no-cache --virtual .build-deps \
-        build-base linux-headers \
-        ruby-dev gnupg \
- && echo 'gem: --no-document' >> /etc/gemrc \
- && gem install oj -v 3.13.22 \
- && gem install json -v 2.6.2 \
- && gem install async -v 1.30.3 \
- && gem install async-http -v 0.56.6 \
+# therefore an 'apt-get purge' has no effect
+RUN buildDeps=" \
+      make gcc g++ libc-dev \
+      wget bzip2 gnupg dirmngr\
+    " \
  && gem install fluentd -v 1.17.0 \
- && gem install bigdecimal -v 1.4.4 \
-# NOTE: resolv v0.2.1 includes the fix for CPU spike issue due to DNS resolver.
-# This hack is needed for Ruby 2.6.7, 2.7.3 and 3.0.1. (alpine image is still kept on 2.7.3)
- && gem install resolv -v 0.2.1 \
- && gem install elasticsearch -v 7.13.3 \
- && gem install elasticsearch-xpack -v 7.13.3 \
- && gem install fluent-plugin-detect-exceptions -v 0.0.14 \
- && gem install \
-         fluent-plugin-s3 \
```

Comment:
the built-in fluentd plugins shouldn't be remove I think

Reply:
@wanjunlei @wenchajun what do you think?

Reply:
in the latest fluentd build workflow:
- fluentd image is actually first build and pushed to ghcr, not dockerhub https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L252-L268
https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L294-L310
https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L471-L472
- And after all the arm64 and amd64 fluentd images are built and push to ghcr, all these images are synced to dockerhub kubespere org https://github.com/fluent/fluent-operator/blob/master/.github/workflows/build-fd-image.yaml#L494-L505

Using a image on dockerhub kubesphere org as base image is not ok in my opinion

Reply:
Sorry, this is a modification I didn't realize. I made a mistake when I copied it.